### PR TITLE
fix(sheet): cell related facade api

### PR DESCRIPTION
**Pull request by Dushusir**
## What's updated?
Facade API
1. - [x] `onCellRender`
2. - [x] `refreshCanvas`
3. - [x] `onBeforeCellEdit`
4. - [x] `onAfterCellEdit`

Docs
Write API documentation after the new documentation system is built
## How to test?
Testing with Uniscript

1. Test `onCellRender` and `refreshCanvas`

Case 1, fixed position rendering, adding rows will not affect the rendering position
```js
// Fixed position rendering
univerAPI.getSheetHooks().onCellRender([{
    drawWith: (ctx, info, skeleton, spreadsheets)=>{
        const {row, col} = info;
        // Update to any cell location you want
        if(row === 1 && col === 1){
            const { primaryWithCoord } = info;
            const {startX, startY} = primaryWithCoord
            ctx.fillText('✅', startX, startY+10)
        }
    }
}])

// refresh canvas
univerAPI.getActiveWorkbook().getActiveSheet().refreshCanvas()
```
Case 2, rendering by markup, the mark position will change with the row and column, and the rendering position will also change accordingly
```js
// Mark in advance
univerAPI.getActiveWorkbook().getActiveSheet().getRange('B2').setValue({custom:{key:'needCheck'}})
// Rendering by markup
univerAPI.getSheetHooks().onCellRender([{
    drawWith: (ctx, info, skeleton, spreadsheets)=>{
        const {row, col, data} = info;
        // Update to any cell location you want
        if(data?.custom?.key === 'needCheck'){
            const { primaryWithCoord } = info;
            const {startX, startY} = primaryWithCoord
            ctx.fillText('✅', startX, startY+10)
        }
    }
}])

// refresh canvas
univerAPI.getActiveWorkbook().getActiveSheet().refreshCanvas()
```
2. Test `onBeforeCellEdit`

```js
univerAPI.getSheetHooks().onBeforeCellEdit((params)=>{
    console.info('cell edit before',params)
})
```
3. Test `onAfterCellEdit`
```js
univerAPI.getSheetHooks().onAfterCellEdit((params)=>{
    console.info('cell edit after',params)
})
```

## Pull Request Checklist

- [ ] Related tickets or issues have been linked in the PR description (or missing issue).
- [ ] [Naming convention](https://github.com/dream-num/univer/blob/dev/docs/NAMING_CONVENTION.md) is followed (**do please** check it especially when you created new plugins, commands and resources).
- [ ] Unit tests have been added for the changes (if applicable).
- [ ] Breaking changes have been documented (or no breaking changes introduced in this PR).

### CODE REVIEW

**Review by hexf00 — APPROVED** [review]

File: packages/sheets-ui/src/facade/f-worksheet.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+     * Refresh the canvas.
+     */
+    refreshCanvas(): void;
+}
+
+class FWorksheetSkeletonMixin extends FWorksheet implements IFWorksheetSkeletonMixin {
+    override refreshCanvas(): void {
+        const renderManagerService = this._injector.get(IRenderManagerService);
+        const unitId = this._fWorkbook.id;
+        renderManagerService.getRenderById(unitId)?.with(SheetSkeletonManagerService).reCalculate();
```
Comment (wzhudev) [inline packages/sheets-ui/src/facade/f-worksheet.ts:33]: 这里如果没有发现对应的 renderer，抛出错误比较好，因为这个是给最终用户用的 API

File: packages/sheets-ui/src/facade/f-worksheet.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        renderManagerService.getRenderById(unitId)?.mainComponent?.makeDirty();
```
Comment (wzhudev) [inline packages/sheets-ui/src/facade/f-worksheet.ts:34]: 这里 `getRenderById()` 的结果应该缓存为局部变量

File: packages/sheets-ui/src/facade/f-sheet-hooks.ts [hunk 3]
```diff
@@ -63,4 +65,55 @@ export class FSheetHooks {
     onCellDrop(callback: (cellPos: Nullable<IDragCellPosition>) => void): IDisposable {
         return toDisposable(this._dragManagerService.endCell$.subscribe(callback));
     }
+
+    /**
+     * The onCellRender event is fired when a cell is rendered.
+     * @param customRender
+     * @param effect
+     * @param priority
+     * @returns
```
Comment (wzhudev) [inline packages/sheets-ui/src/facade/f-sheet-hooks.ts:74]: 这里最好对 `@param` `@returns` 做一下具体描述

**Review by wzhudev — COMMENTED** [review]
LGTM. Some usability details need to be addressed.

**Comment by wzhudev** [comment]
I am marking this PR as verified because it only changes API.

File: packages/sheets-ui/src/facade/f-worksheet.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+class FWorksheetSkeletonMixin extends FWorksheet implements IFWorksheetSkeletonMixin {
+    override refreshCanvas(): void {
+        const renderManagerService = this._injector.get(IRenderManagerService);
+        const unitId = this._fWorkbook.id;
+        const render = renderManagerService.getRenderById(unitId);
+
+        if (!render) {
+            throw new Error('Render not found');
```
Comment (wzhudev) [inline packages/sheets-ui/src/facade/f-worksheet.ts:36]: Please consider to expose more debug info in error messages. For example, `unitId`.

**Review by wzhudev — APPROVED** [review]
LGTM

File: packages/sheets/src/index.ts [hunk 5]
```diff
@@ -177,6 +177,7 @@ export { getInsertRangeMutations, getRemoveRangeMutations } from './commands/uti
 export { handleInsertRangeMutation } from './commands/utils/handle-range-mutation';
 export { type ISheetCommandSharedParams } from './commands/utils/interface';
 export { copyRangeStyles } from './commands/commands/utils/selection-utils';
+export { FSheetHooks } from './facade/f-sheet-hooks';
```
Comment (hexf00) [inline packages/sheets/src/index.ts:180]: move to facade.ts

**Review by hexf00 — CHANGES_REQUESTED** [review]

File: packages/sheets-ui/src/facade/f-sheet-hooks.ts [hunk 6]
```diff
@@ -14,53 +14,119 @@
...
-        @Inject(Injector) private readonly _injector: Injector,
-        @Inject(HoverManagerService) private readonly _hoverManagerService: HoverManagerService,
-        @Inject(DragManagerService) private readonly _dragManagerService: DragManagerService
-    ) {
-        // empty
-    }
+import type { ICellCustomRender, IDisposable, Nullable } from '@univerjs/core';
+import type { IDragCellPosition, IEditorBridgeServiceVisibleParam, IHoverCellPosition } from '@univerjs/sheets-ui';
+import { ICommandService, InterceptorEffectEnum, toDisposable } from '@univerjs/core';
+import { FSheetHooks, InterceptCellContentPriority, INTERCEPTOR_POINT, SheetInterceptorService } from '@univerjs/sheets';
```
Comment (hexf00) [inline packages/sheets-ui/src/facade/f-sheet-hooks.ts:20]: import { FSheetHooks } from '@univerjs/sheets/facade';

**Review by hexf00 — CHANGES_REQUESTED** [review]

**Review by hexf00 — APPROVED** [review]

**Pull request merged**